modules/nf-core/seurat/multiseq/templates/MULTIseq.R

```diff
@@ -0,0 +1,197 @@
+################################################
+################################################
+## Generate plots                             ##
+################################################
+################################################
+
+if(opt\$produce_plots){
+        ### Ridge plot
+        Idents(demultiplex) <- opt_args_transformed\$group_cells_feature_scatter
+        RidgePlot(demultiplex, assay = opt_args_transformed\$assay , features = rownames(demultiplex[[opt_args_transformed\$assay ]])[1:opt_args_transformed\$number_of_features_ridge_plot], ncol = opt_args_transformed\$number_of_cols_ridge_plot)
+        ggsave(paste0(opt\$output_prefix, '_ridge_plot.jpeg'), device = "jpeg", dpi = 500)
+
+        ### Feature scatter plot
+        FeatureScatter(demultiplex, feature1 = opt_args_transformed\$feature_scatter_feature_1, feature2 = opt_args_transformed\$feature_scatter_feature_2)
+        ggsave(paste0(opt\$output_prefix, '_feature_scatter_plot.jpeg'), device = "jpeg", dpi = 500)
+
+        ### Violin plot
+        Idents(demultiplex) <- opt_args_transformed\$group_cells_violin_plot
+        VlnPlot(demultiplex, features = opt_args_transformed\$features_violin_plot, pt.size = opt_args_transformed\$pt_size, log = opt_args_transformed\$log)
+        ggsave(paste0(opt\$output_prefix, '_violin_plot.jpeg'), device = "jpeg", dpi = 500)
+
+        ### TSNE plot for HTOs
+        subset_demultiplexing_results <- subset(demultiplex, idents = opt_args_transformed\$subset_idents, invert = opt_args_transformed\$subset_invert)
+        DefaultAssay(subset_demultiplexing_results) <- opt_args_transformed\$assay
+        subset_demultiplexing_results <-ScaleData(subset_demultiplexing_results, features = rownames(subset_demultiplexing_results),
+        verbose = opt_args_transformed\$tsne_scale_data_verbose)
+        subset_demultiplexing_results <- RunPCA(subset_demultiplexing_results, features =  rownames(subset_demultiplexing_results), approx = opt_args_transformed\$run_pca_approx)
+        subset_demultiplexing_results <- RunTSNE(subset_demultiplexing_results, dims = 1:opt_args_transformed\$run_tsne_dim_max, perplexity = opt_args_transformed\$run_tsne_perplexity, check_duplicates = opt_args_transformed\$check_duplicates_tsne)
+        DimPlot(subset_demultiplexing_results)
+        ggsave(paste0(opt\$output_prefix, '_tsne_plot.jpeg'), device = "jpeg", dpi = 500)
+
+        ### TSNE plot classification
+        hto_names <- rownames(demultiplex[[opt_args_transformed\$assay]])
+        # Extract the singlets
+        pbmc.singlet <- subset(demultiplex, idents = hto_names)
```

Comment:
Why `pbmc`?

Reply:
it was the default name used in the tutorial to create the plots given by the library. I just renamed the variable to something more understandable.

Reply:
PBMC stands for Peripheral Blood Mononuclear Cells, i.e. a type of white blood cells. They are a common sample type which are easy to get hold of, presumably that is the dataset used in the tutorial. Could you make it more generic?

Reply:
Yes, I changed the name of the variable in last update